docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,197 @@
+
+## Proposal
+
+This proposal will consist of different steps:
+
+1. Remove the `spec.resources` from the `Build` and `BuildRun` API.
+
+2. Remove the logic inside the `generate_taskrun.go` and test classes, where we parse resources defined in the `Build/BuildRun` into the strategy containers. Ensure `resources` of the strategies are properly used.
+
+3. For all the current strategies(_BuildStrategy or ClusterBuildStrategy_) in this repository (_samples_), we should define default values for the container resources limits/request so that our `travis` will not be affected by strategy containers consuming all the CPU and memory (_unlimited_). Also, update all of our sample `Build` YAMLs to use this strategies.
+
+4. Properly document under the strategies docs, that all of the steps resources can be modified by the `strategy admin`. If a difference of values per strategy is needed, then multiple `ClusterBuildStrategy` will be required to exist on the cluster, with different names.
+
+5. Document different flavours for each `ClusterBuildStrategy` (_e.g. kaniko, buildpacks_). Each flavour should carefully illustrate how it differentiate with the default `ClusterBuildStrategy` values, and which advantage one can have when choosing this flavour under their `Build` resource. Advantages should be for now `faster builds`.
+
+To illustrate these changes from the perspective of the end-user, lets take a look on the default scenario:
+
+```yaml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: buildpack-nodejs-build
+  annotations:
+    build.build.dev/build-run-deletion: "false"
+spec:
+  source:
+    url: https://github.com/sclorg/nodejs-ex
+  strategy:
+    name: buildpacks-v3-default <---- PAY ATTENTION TO THIS DEFAULT STRATEGY FLAVOUR
+    kind: ClusterBuildStrategy
+....
+```
+
+If an `end-user` will require a different flavour for `buildpacks` strategies because he/she wants faster builds, then the `Build` definition will look like:
+
+```yaml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: buildpack-nodejs-build
+  annotations:
+    build.build.dev/build-run-deletion: "false"
+spec:
+  source:
+    url: https://github.com/sclorg/nodejs-ex
+  strategy:
+    name: buildpacks-v3-medium <---- PAY ATTENTION TO THIS MEDIUM STRATEGY FLAVOUR
+    kind: ClusterBuildStrategy
+....
+```
+
+If the same `end-user` is willing to get more resources in order to even go beyond the previous build times, he/she could select a different flavour for buildpacks, see the following:
+
+```yaml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: buildpack-nodejs-build
+  annotations:
+    build.build.dev/build-run-deletion: "false"
+spec:
+  source:
+    url: https://github.com/sclorg/nodejs-ex
+  strategy:
+    name: buildpacks-v3-large <---- PAY ATTENTION TO THIS LARGE STRATEGY FLAVOUR
+    kind: ClusterBuildStrategy
+....
+```
+
```

Comment:
lots of good yaml examples here ... that said, should we add a yaml sample here that shows the proposed way of setting the resources on the Strategy object (like was illustrated on the community meeting today) ?

Reply:
@gabemontero of course, let me update the PR

Reply:
perfect @qu1queee thanks